source/declarations.tex

```diff
@@ -4239,3 +4240,414 @@
+in functions invoked from that predicate.
+An evaluation of a predicate that exits via an exception
+invokes \tcode{std::terminate()}\iref{except.terminate}.
+The behavior of any other side effect is undefined.
+\begin{example}
+\begin{codeblock}
+void push(int x, queue & q)
+  [[expects: !q.full()]]
+  [[ensures: !q.empty()]]
+{
+  @\commentellip@
+  [[assert: q.is_valid()]];
+  @\commentellip@
+}
+
+int min = -42;
+constexpr int max = 42;
+
+constexpr int g(int x)
+  [[expects: min <= x]]                         // error
+  [[expects: x < max]]                          // OK
+{
+  @\commentellip@
+  [[assert: 2*x < max]];
+  [[assert: ++min > 0]];                        // undefined behavior
+  @\commentellip@
+}
+\end{codeblock}
+\end{example}
+
+\pnum
+If a postcondition odr-uses\iref{basic.def.odr}
+a parameter value in its predicate
+and the function body makes direct or indirect modifications of that value,
+the behavior is undefined.
+\begin{example}
+\begin{codeblock}
+int f(int x)
+  [[ensures r: r == x]]
+{
+  return ++x;                   // undefined behavior
+}
+
+int g(int * p)
+  [[ensures r: p != nullptr]]
+{
+  *p = 42;                      // OK, \tcode{p} is not modified
+}
+
+int h(int x)
+  [[ensures r: r == x]]
+{
+  potentially_modify(x);        // undefined behavior if \tcode{x} is modified
+  return x;
+}
+\end{codeblock}
+\end{example}
+
+\pnum
+If the \grammarterm{contract-level}
+of a \grammarterm{contract-attribute-specifier} is absent,
+it is assumed to be \tcode{default}.
+\begin{note}
+A \tcode{default} \grammarterm{contract-level} is expected to be used
+for those contracts where
+the cost of run-time checking is assumed to be
+small (or at least not expensive) compared to
+the cost of executing the function.
+An \tcode{audit} \grammarterm{contract-level} is expected to be used
+for those contracts where
+the cost of run-time checking is assumed to be
+large (or at least significant) compared to
+the cost of executing the function.
+An \tcode{axiom} \grammarterm{contract-level} is expected to be used
+for those contracts that are formal comments
+and are not evaluated at run-time.
+\end{note}
+
+\pnum
+Multiple contract conditions may be applied to a function type
+with the same or different \grammarterm{contract-level}{s}.
+\begin{example}
+\begin{codeblock}
+int z;
+
+bool is_prime(int k);
+
+void f(int x)
+  [[expects: x > 0]]
+  [[expects audit: is_prime(x)]]
+  [[ensures: z > 10]]
+{
+  @\commentellip@
+}
+\end{codeblock}
+\end{example}
+
+\pnum
+A translation may be performed
+with one of the following \defnx{build levels}{build level}:
+\term{off}, \term{default}, or \term{audit}.
+A translation with build level set to \term{default}
+performs checking for \tcode{default} contracts.
+A translation with build level set to \term{audit}
+performs checking for \tcode{default} and \tcode{audit} contracts.
+If no build level is explicitly selected, the build level is \term{default}.
```

Comment:
We should explicitly say that the build level *off* means checking is not performed for any contracts, rather than leaving the reader wondering.

Reply:
Fixed.